app/models/UserAnswers.scala

```diff
@@ -44,7 +50,12 @@ final case class UserAnswers(
                               internalAuthId :String
                             ) extends ReadableUserAnswers {
 
-  def set[A](page: Settable[A], value: A)(implicit writes: Writes[A]): Try[UserAnswers] = {
+  def set[A](page: Settable[A], value: A)(implicit writes: Writes[A], reads: Reads[A]): Try[UserAnswers] = {
+
+    val hasValueChanged: Boolean = getAtPath(page.path) match {
```

Comment:
Could this be something like:

val hasValueChanged: Boolean = getAtPath(page.path).contains(value)